eclair-core/src/main/scala/fr/acinq/eclair/Features.scala

```diff
@@ -154,4 +161,67 @@ object Features {
    */
   def areSupported(features: ByteVector): Boolean = areSupported(features.bits)
 
+  object Resolution {
```

Comment:
I think the functions/contracts in this object are a bit confusing.
What do you think of the following alternative proposal:

```scala
object Resolution {
    
    private val knownFeatures: Seq[Feature] = Seq(
      OptionDataLossProtect,
      InitialRoutingSync,
      ChannelRangeQueries,
      VariableLengthOnion,
      ChannelRangeQueriesExtended,
      PaymentSecret,
      BasicMultiPartPayment,
      Wumbo,
      TrampolinePayment
    )

    case class ActivatedFeature(feature: Feature, support: FeatureSupport)

    def fromConfiguration(config: Config): Seq[ActivatedFeature] = ???
    
    def toByteVector(features: Seq[ActivatedFeature]): ByteVector = ???

    def fromByteVector(features: ByteVector): Seq[ActivatedFeature] = ???

}
```

The `ActivatedFeature` would replace the `HumanReadableFeature` in `Eclair.scala`.
We could maybe even create `case class ActivatedFeatures(features: ActivatedFeature*)` and put `fromByteVector/toByteVector` as methods on this class? I think that would make the contracts clearer.

Reply:
Currently the role of `HumanReadableFeature` is to help render the features as JSON, notice how the fields `name` and `support` end up serialized as JSON and that's what i wanted to achieve, it makes sense for anyone consuming the API to have that kind of object in the response.

I like the suggestion of introducing `ActivatedFeature` but i think it's even better if we take it one step further and incapsulate them inside a `Features` object, this would be carried around in the codebase instead of the ByteVector features.

```
case class Features(features: Seq[ActivatedFeature]){
  def hasFeature(f: Feature): Boolean
  def toByteVector: ByteVector
  def apply(raw: ByteVector): Features
}
```
WDYT?

Reply:
> how the fields name and support end up serialized as JSON

But that's true as well with `ActivatedFeature`, right? The only difference is that the JSON is `{ feature: data_loss_protect, support: mandatory }` instead of `name`?

> I like the suggestion of introducing ActivatedFeature but i think it's even better if we take it one step further and incapsulate them inside a Features object, this would be carried around in the codebase instead of the ByteVector features.

I really like that! That means `NodeParams` would directly contain this `Features` object instead of a `ByteVector`, it would be really nice.

Reply:
Let's see what the `ActivatedFeature` idea lead us to, but imo this `Resolution` class is a *lot* of code for what it's doing.